examples/src/bin/unbuffered-server.rs

```diff
@@ -0,0 +1,255 @@
+                    }
+
+                    received_request = true;
+                }
+            }
+
+            ConnectionState::MustEncodeTlsData(mut state) => {
+                try_or_resize_and_retry(
+                    |out_buffer| state.encode(out_buffer),
+                    |e| {
+                        if let EncodeError::InsufficientSize(is) = &e {
+                            Ok(*is)
+                        } else {
+                            Err(e.into())
+                        }
+                    },
+                    outgoing_tls,
+                    &mut outgoing_used,
+                )?;
+            }
+
+            ConnectionState::MustTransmitTlsData(state) => {
+                send_tls(&mut sock, outgoing_tls, &mut outgoing_used)?;
+                state.done();
+            }
+
+            ConnectionState::NeedsMoreTlsData { .. } => {
+                recv_tls(&mut sock, incoming_tls, &mut incoming_used)?;
+            }
+
+            ConnectionState::TrafficTransit(mut state) => {
+                if !received_request {
+                    recv_tls(&mut sock, incoming_tls, &mut incoming_used)?;
+                } else {
+                    let map_err = |e| {
+                        if let EncryptError::InsufficientSize(is) = &e {
+                            Ok(*is)
+                        } else {
+                            Err(e.into())
+                        }
+                    };
+
+                    let http_response = build_http_response();
+                    try_or_resize_and_retry(
+                        |out_buffer| state.encrypt(http_response, out_buffer),
+                        map_err,
+                        outgoing_tls,
+                        &mut outgoing_used,
+                    )?;
+                    sent_response = true;
+
+                    try_or_resize_and_retry(
+                        |out_buffer| state.queue_close_notify(out_buffer),
+                        map_err,
+                        outgoing_tls,
+                        &mut outgoing_used,
+                    )?;
+                    open_connection = false;
+
+                    send_tls(&mut sock, outgoing_tls, &mut outgoing_used)?;
+                }
+            }
+
+            _ => unreachable!(),
+        }
+
+        if discard != 0 {
+            assert!(discard <= incoming_used);
+
+            incoming_tls.copy_within(discard..incoming_used, 0);
+            incoming_used -= discard;
+
+            eprintln!("discarded {discard}B from `incoming_tls`");
+        }
+
+        iter_count += 1;
+        assert!(
+            iter_count < MAX_ITERATIONS,
+            "did not get a HTTP response within {MAX_ITERATIONS} iterations"
+        );
+    }
+
+    assert!(received_request);
+    assert!(sent_response);
+    assert_eq!(0, incoming_used);
+    assert_eq!(0, outgoing_used);
+
+    Ok(())
+}
+
+fn try_or_resize_and_retry<E>(
```

Comment:
I think this example should be using the helpers pulled out in 50911be8a80e2b6504b295516d86a9c3112e361f right?

Reply:
What happened to the commit that pulled out the helpers? It looks like it got dropped from this branch and so this feedback remains unresolved.

Reply:
Explained by Ctz: this was removed to make the examples more stand-alone, which makes sense to me!